src/algebra/big_operators/basic.lean

```diff
@@ -395,6 +377,35 @@ begin
     simp only [prod_insert H, prod_mul_distrib, ih] }
 end
 
+@[to_additive]
+lemma prod_product_left {s : finset γ} {t : finset α} {f : γ×α → β} :
+  (∏ x in s.product t, f x) = ∏ x in s, ∏ y in t, f (x, y) :=
+begin
+  haveI := classical.dec_eq α, haveI := classical.dec_eq γ,
+  rw [product_eq_bUnion, prod_bUnion],
+  { congr, funext, exact prod_image (λ _ _ _ _ H, (prod.mk.inj H).2) },
+  simp only [disjoint_iff_ne, mem_image],
+  rintros _ _ _ _ h ⟨_, _⟩ ⟨_, _, ⟨_, _⟩⟩ ⟨_, _⟩ ⟨_, _, ⟨_, _⟩⟩ _,
+  apply h, cc
+end
+
+/-- An uncurried version of `finset.prod_product_left`. -/
+@[to_additive "An uncurried version of `finset.sum_product_left`"]
+lemma prod_product_left' {s : finset γ} {t : finset α} {f : γ → α → β} :
+  (∏ x in s.product t, f x.1 x.2) = ∏ x in s, ∏ y in t, f x y :=
+prod_product_left
+
+@[to_additive]
+lemma prod_product_right {s : finset γ} {t : finset α} {f : γ×α → β} :
+  (∏ x in s.product t, f x) = ∏ y in t, ∏ x in s, f (x, y) :=
+by rw [prod_product_left, prod_comm]
+
+/-- An uncurried version of `finset.prod_product_right`. -/
+@[to_additive "An uncurried version of `finset.prod_product_right`"]
+lemma prod_product_right' {s : finset γ} {t : finset α} {f : γ → α → β} :
+  (∏ x in s.product t, f x.1 x.2) = ∏ y in t, ∏ x in s, f x y :=
+prod_product_right
```

Comment:
This PR doesn't even use these new lemmas! Are you sure they're useful vs just having the user `rw prod_comm` first?

Reply:
Eh! `WIP` isn't there for nothing 😉 They're used now.

Reply:
> They're used now.

Where?

Reply:
> My idea is that products should be handled symmetrically, so the API shouldn't favor the left side.

It feels a little bit to me like having a lemma that sends `a * (b + c)` to `a * b + a * c` but not having a lemma that sends it to  `b * a + c * a`; but maybe I'll come up with a better example later.

Reply:
Uh, more like not having `(b + c) * a = b * a + c * a`.

Reply:
Okay, I've reverted it all. I still don't think you're making a point, though.